tool/embedded-binaries.ts

```diff
@@ -70,162 +71,216 @@ export async function getEmbeddedProtocol(outPath: string): Promise<void> {
+        `-${getOs()}-${getArch()}` +
+        getArchiveFileExtension(),
+      outPath,
     });
-    shell.exec('git reset --hard FETCH_HEAD', {silent: true, cwd: repoPath});
+    fs.rename(p.join(outPath, 'sass_embedded'), p.join(outPath, repo));
   } else {
-    console.log('Cloning Dart Sass Embedded.');
-    shell.exec(
-      `git clone \
-        --depth=1 \
-        git://github.com/sass/dart-sass-embedded \
-        ${repoPath}`,
-      {silent: true}
-    );
-  }
-
-  console.log('Downloading Dart Sass Embedded dependencies.');
-  shell.exec('pub upgrade', {silent: true, cwd: repoPath});
-
-  console.log('Building Dart Sass Embedded.');
-  shell.exec('pub run grinder protobuf pkg-standalone-dev', {
-    silent: true,
-    cwd: repoPath,
-  });
-
-  if (getOs() === 'windows') {
-    shell.cp('-R', p.join(repoPath, 'build'), p.join(outPath, 'sass_embedded'));
-  } else {
-    // Symlinking doesn't play nice with Jasmine's test globbing on Windows.
-    fs.symlink(
-      p.relative(outPath, p.join(repoPath, 'build')),
-      p.join(outPath, 'sass_embedded')
-    );
+    fetchRepo(repo, BUILD_PATH, options.version);
+    // TODO(awjin): Allow building and linking from custom path for local dev.
+    buildDartSassEmbedded(p.join(BUILD_PATH, repo));
+    await linkBuiltFiles(repo, p.join(BUILD_PATH, repo, 'build'), outPath);
   }
 }
 
-// Downloads the latest release version of the Embedded Dart Sass compiler.
-async function downloadDartSassEmbedded(
-  outPath: string,
-  version?: string
-): Promise<void> {
-  await cleanEmbeddedDir(outPath);
-
-  let latestRelease: {
-    tag_name: string;
-    name: string;
+// Downloads release asset into `outPath`. Returns the version that was
+// downloaded.
+async function downloadRelease(options: {
+  repo: string;
+  outPath: string;
+  version?: string;
+  assetUrl: (release: ReleaseInfo) => string;
+  tags?: boolean;
+}): Promise<string> {
+  const fetchOptions: RequestInit = {
+    redirect: 'follow',
   };
 
-  // TODO(awjin): Once dart-sass-embedded is no longer under pre-release, call
-  // the Github API instead of manually piecing together the asset URL. The
-  // API exposes only the latest *non*-pre-release versions.
-  // https://docs.github.com/en/free-pro-team@latest/rest/reference/repos#get-the-latest-release
-  console.log('Getting Dart Sass Embedded release info.');
+  console.log(`Negotiating version for ${options.repo}.`);
+  let releaseInfo: ReleaseInfo | undefined;
   try {
-    const fetchOptions: RequestInit = {
-      redirect: 'follow',
-    };
-    if (process.env.TRAVIS === 'true' && process.env.GITHUB_AUTH) {
-      fetchOptions['headers'] = {
-        Authorization:
-          'Basic ' +
-          Buffer.from(`sassbot:${process.env.GITHUB_AUTH}`).toString('base64'),
-      };
-    }
+    const response = await fetch(
+      `https://api.github.com/repos/sass/${options.repo}/${
+        options.tags ? 'tags' : 'releases'
+      }`,
```

Comment:
```suggestion
      `https://api.github.com/repos/sass/${options.repo}/` +
      `${options.tags ? 'tags' : 'releases'}`,
```

Style nit: I think it's usually clearer to have a string split across multiple lines with `+` rather than having a multiline interpolation block.